`broadcast_tx_commit` behavior changed between 1.19.2 and 1.20.0-rc.1

The latest release broke `backward_compatible.py` because the behavior of `broadcast_tx_commit` changed. More specifically, https://github.com/near/nearcore/blob/b7c8a3b5d2bfaeb1d689972fd867d1c10afcc440/pytest/tests/sanity/backward_compatible.py#L110 used to work if the stable node runs 1.19.0 but the assertion fails with 1.20.0-rc.1. What happened is that the test is not well written and the promise actually fails. However, since the promise is not the return result of the function call, the execution is considered successful, but it appears that something changed with 1.20.0-rc.1 and the rpc response becomes
```
AssertionError: {'jsonrpc': '2.0', 'error': {'name': 'HANDLER_ERROR', 'cause': {'name': 'TIMEOUT_ERROR'}, 'code': -32000, 'message':
'Server error', 'data': 'Timeout'}, 'id': 'dontcare'}
```
instead.

To reproduce, change [`branch`](https://github.com/near/nearcore/blob/b7c8a3b5d2bfaeb1d689972fd867d1c10afcc440/pytest/tests/sanity/backward_compatible.py#L28) between `1.20.0` and `1.19.2` to see the difference.

--- a/neard/src/cli.rs
+++ b/neard/src/cli.rs
@@ -311,7 +311,7 @@ pub(super) struct TestnetCmd {
     #[clap(long, default_value = "node")]
     prefix: String,
     /// Number of shards to initialize the testnet with.
-    #[clap(long, default_value = "4")]
+    #[clap(long, default_value = "1")]
     shards: NumShards,
     /// Number of validators to initialize the testnet with.
     #[clap(long = "v", default_value = "4")]

